engine/src/dskw32.cpp

```diff
@@ -1355,13 +1373,61 @@ struct MCStdioFileHandle: public MCSystemFileHandle
 
 	virtual bool Write(const void *p_buffer, uint32_t p_length)
 	{
-		uint32_t t_written;
 		if (this == IO_stdin)
 			return true; // Shouldn't it return false???
 
 		if (m_handle == NULL)
 			return false;
 
+		/* If this is not a pipe then we may have a write buffer. */
+		if (!m_is_pipe)
```

Comment:
@runrevmark I think you need `&& m_should_buffer_writes` here ?

Reply:
Oh! I can simplify that - I can set should_buffer_writes to false if the stream is a pipe. Thanks!

Reply:
@montegoulding That should be addressed now - I've changed this if to `m_should_buffer_writes`, and only set the that to true if not a pipe in the constructor.